### PR TITLE
[Pages] Add improvements and troubleshooting guide to git-integration.md

### PR DESCRIPTION
**Pull request by cmsparks**
* Pretty straightforward edits to the git-integration.md section of the Pages documentation

### CODE REVIEW

File: content/pages/platform/git-integration.md [hunk 1]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+3. Go back to the `Workers & Pages` overview page at `https://dash.cloudflare.com/[YOUR\_ACCOUNT\_ID]/workers-and-pages`. Click `Create application` > `Pages` > `Connect to git`
+4. Select the GitLab tab at the top, click the '+ Add account' button, click the GitLab account you want to add, and then click `Authorize` on the modal titled "Authorize Cloudflare Pages to use your account? ".
+5. You should be redirected to the create project page with your GitHub account or organization in the account list.
+6. Attempt to make a new deployment with your project which was previously broken.
+
+## Troubleshooting
+
+If you run into any issues related to deployments failing, check your project dashboard to see if there are any SCM installation warnings listed. To resolve any errors displayed in the Cloudflare Pages dashboard, follow the steps listed below.
+
+### `The Cloudflare Pages installation was uninstalled on GitHub. To reinstall, log in to the Cloudflare dashboard > Workers & Pages > Create an application > Connect to git > + Add account.`
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: - [x] `Connect to Git`

File: content/pages/platform/git-integration.md [hunk 2]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+When encountering Git integration related issues, one potential troubleshooting step is attempting to uninstall and reinstall an installation. The process for each git provider is provided below
+
+### GitHub
+
+1. Go to the installation settings page on GitHub
+    1. `https://github.com/settings/installations` for individual accounts
+    1. `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts
+2. If the Cloudflare Pages installation is there, click `Configure`, and click `Uninstall 'Cloudflare Pages'` (if there is no "Cloudflare Pages" installation, the user doesn't need to do anything)
+3. Go back to the `Workers & Pages` overview page at `https://dash.cloudflare.com/[YOUR\_ACCOUNT\_ID]/workers-and-pages`. Click `Create application` > `Pages` > `Connect to git`
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: - [x] `Connect to Git`

File: content/pages/platform/git-integration.md [hunk 3]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
 ![Pausing a deployment in the Settings of your Pages project](/images/pages/platform/git.pause.png)
 
 Selecting **Pause deployments** will present a confirmation and, once confirmed, the **Pause deployments** button will be replaced with a **Resume deployments** button. While paused, your **Deployments** list will present a banner message, reminding you that automatic deployments are not enabled.
+
+## Reinstalling a git installation
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: - [x] ... `a Git installation`

File: content/pages/platform/git-integration.md [hunk 4]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+When encountering Git integration related issues, one potential troubleshooting step is attempting to uninstall and reinstall an installation. The process for each git provider is provided below
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: technically you're un/re-installing the app not the installation

Comment (CarmenPopoviciu) [reply content/pages/platform/git-integration.md]: - [x] `The process for each Git provider`...

File: content/pages/platform/git-integration.md [hunk 5]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+### GitHub
+
+1. Go to the installation settings page on GitHub
+    1. `https://github.com/settings/installations` for individual accounts
+    1. `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts
+2. If the Cloudflare Pages installation is there, click `Configure`, and click `Uninstall 'Cloudflare Pages'` (if there is no "Cloudflare Pages" installation, the user doesn't need to do anything)
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: we should use the same quote convention when spelling Cloudflare Pages, meaning either both with "" or both without

File: content/pages/platform/git-integration.md [hunk 6]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+3. Go back to the `Workers & Pages` overview page at `https://dash.cloudflare.com/[YOUR\_ACCOUNT\_ID]/workers-and-pages`. Click `Create application` > `Pages` > `Connect to git`
+4. Click the '+ Add account' button, click the GitHub account you want to add, and then click `Install & Authorize`.
+5. You should be redirected to the create project page with your GitHub account or organization in the account list.
+6. Attempt to make a new deployment with your project which was previously broken.
+
+### GitLab
+
+1. Go to your application settings page on GitLab located here: https://gitlab.com/-/profile/applications
+2. Click the "Revoke" button on your Cloudflare Pages installation if it exists.
+3. Go back to the `Workers & Pages` overview page at `https://dash.cloudflare.com/[YOUR\_ACCOUNT\_ID]/workers-and-pages`. Click `Create application` > `Pages` > `Connect to git`
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: - [ ] `Connect to Git`

File: content/pages/platform/git-integration.md [hunk 7]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+4. Select the GitLab tab at the top, click the '+ Add account' button, click the GitLab account you want to add, and then click `Authorize` on the modal titled "Authorize Cloudflare Pages to use your account? ".
+5. You should be redirected to the create project page with your GitHub account or organization in the account list.
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md:70]: - [ ] `GitLab` not `GitHub`

File: content/pages/platform/git-integration.md [hunk 8]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+6. Attempt to make a new deployment with your project which was previously broken.
+
+## Troubleshooting
+
+If you run into any issues related to deployments failing, check your project dashboard to see if there are any SCM installation warnings listed. To resolve any errors displayed in the Cloudflare Pages dashboard, follow the steps listed below.
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: should we add a screenshot here with an example of such a warning?

File: content/pages/platform/git-integration.md [hunk 9]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+### `The Cloudflare Pages installation was uninstalled on GitHub. To reinstall, log in to the Cloudflare dashboard > Workers & Pages > Create an application > Connect to git > + Add account.`
+
+To resolve this issue, please follow the steps provided above in the [Reinstalling a git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, please contact support.
+
+### `Cloudflare Pages is not properly installed on your git account. Please uninstall and reinstall your git installation to make new deployments.`
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: - [x] `Git account`

- [x] `Git installation`

File: content/pages/platform/git-integration.md [hunk 10]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+### `The Cloudflare Pages installation has been suspended. Please unsuspend in your GitHub installation settings.`
+
+Go to your GitHub installation settings located at:
+
+* `https://github.com/settings/installations` for individual accounts
+* `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts
+
+Click "Configure" on the Cloudflare Pages application. Scroll down to the bottom of the page and click "Unsuspend" to allow Cloudflare Pages to make future deployments.
+
+### `You have no linked repositories matching the provided name. Please verify this repository exists on the correct git account.`
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: - [x] `Git account`

File: content/pages/platform/git-integration.md [hunk 11]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+### `This repo is unable to be accessed. Please review your GitHub repo access permissions for the Cloudflare Pages installation to make new deployments.`
+
+You may have excluded this repository from your installation's repository access settings.  Go to your GitHub installation settings located at:
+
+* `https://github.com/settings/installations` for individual accounts
+* `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts
+
+Click "Configure" on the Cloudflare Pages application. Under "Repository access," ensure that the repository associated with your Cloudflare Pages project is included in the list.
+
+### `There is an issue with your Cloudflare Pages git installation. Please contact support.`
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: - [x] `Git installation`

File: content/pages/platform/git-integration.md [hunk 12]
```diff
@@ -45,3 +45,65 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+This is an internal error in the Cloudflare Pages SCM system. You can attempt to [reinstalling your git installation](/pages/platform/git-integration/#reinstalling-a-git-installation), but if the issue persists please contact support.
```
Comment (CarmenPopoviciu) [inline content/pages/platform/git-integration.md]: You can attempt to `reinstall` ...

**Review by CarmenPopoviciu — CHANGES_REQUESTED** [review]

File: content/pages/platform/git-integration.md (hunk 8 above)
Comment (cmsparks) [reply content/pages/platform/git-integration.md]: - [x] will add

File: content/pages/platform/git-integration.md [hunk 13]
```diff
@@ -22,11 +22,11 @@ You can deploy projects to Cloudflare Pages from your open-source team, company,
 
 ### GitHub
 
-When authorizing Cloudflare Pages to access your GitHub account, you can also specify access to organizations that you belong to on GitHub.
+When authorizing Cloudflare Pages to access a GitHub account, you can specify access to your individual account or an organization that you belong to on GitHub. In order to be able to add the Cloudflare Pages installation to that organization, your user account must be an owner or have the appropriate role within the organization (i.e. the GitHub Apps Manager role). More information on these roles can be seen on [GitHub's documentation](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization#github-app-managers).
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
When authorizing Cloudflare Pages to access a GitHub account, you can specify access to your individual account or an organization that you belong to on GitHub. In order to be able to add the Cloudflare Pages installation to that organization, your user account must be an owner or have the appropriate role within the organization (that is, the GitHub Apps Manager role). More information on these roles can be seen on [GitHub's documentation](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization#github-app-managers).
```

File: content/pages/platform/git-integration.md [hunk 14]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
 ![Pausing a deployment in the Settings of your Pages project](/images/pages/platform/git.pause.png)
 
 Selecting **Pause deployments** will present a confirmation and, once confirmed, the **Pause deployments** button will be replaced with a **Resume deployments** button. While paused, your **Deployments** list will present a banner message, reminding you that automatic deployments are not enabled.
+
+## Reinstalling a Git installation
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
## Reinstall a Git installation
```

File: content/pages/platform/git-integration.md [hunk 15]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+When encountering Git integration related issues, one potential troubleshooting step is attempting to uninstall and reinstall the GitHub or GitLab application associated with the Cloudflare Pages installation. The process for each Git provider is provided below
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
When encountering Git integration related issues, one potential troubleshooting step is attempting to uninstall and reinstall the GitHub or GitLab application associated with the Cloudflare Pages installation. The process for each Git provider is provided below.
```

File: content/pages/platform/git-integration.md [hunk 16]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+### GitHub
+
+1. Go to the installation settings page on GitHub
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
1. Go to the installation settings page on GitHub:
```

File: content/pages/platform/git-integration.md [hunk 17]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+    1. `https://github.com/settings/installations` for individual accounts
+    1. `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
    1. `https://github.com/settings/installations` for individual accounts.
    1. `https://github.com/organizations/<YOUR_ORGANIZATION_NAME>/settings/installations` for organizational accounts.
```

File: content/pages/platform/git-integration.md [hunk 18]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+2. If the Cloudflare Pages installation is there, click `Configure`, and click `Uninstall "Cloudflare Pages"` (if there is no "Cloudflare Pages" installation, the user doesn't need to do anything)
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
2. If the Cloudflare Pages installation is there, select **Configure**, and select **Uninstall "Cloudflare Pages** (if there is no Cloudflare Pages installation, the user does not need to do anything.)
```

File: content/pages/platform/git-integration.md [hunk 19]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+3. Go back to the `Workers & Pages` overview page at `https://dash.cloudflare.com/[YOUR\_ACCOUNT\_ID]/workers-and-pages`. Click `Create application` > `Pages` > `Connect to Git`
+4. Click the '+ Add account' button, click the GitHub account you want to add, and then click `Install & Authorize`.
+5. You should be redirected to the create project page with your GitHub account or organization in the account list.
+6. Attempt to make a new deployment with your project which was previously broken.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
3. Go back to the Cloudflare dashboard > **Workers & Pages**.
4. Select **Create application** > **Pages** > **Connect to Git**.
5. Select **+ Add account** > select the GitHub account you want to add > **Install & Authorize**.
6. You should be redirected to the create project page with your GitHub account or organization in the account list.
7. Attempt to make a new deployment with your project which was previously broken.
```

File: content/pages/platform/git-integration.md [hunk 20]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+### GitLab
+
+1. Go to your application settings page on GitLab located here: https://gitlab.com/-/profile/applications
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md:66]:
```suggestion
1. Go to your [application settings on GitLab](https://gitlab.com/-/profile/applications).
```

File: content/pages/platform/git-integration.md [hunk 21]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+2. Click the "Revoke" button on your Cloudflare Pages installation if it exists.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md:67]:
```suggestion
2. Select **Revoke** on your Cloudflare Pages installation if it exists.
```

File: content/pages/platform/git-integration.md [hunk 22]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+3. Go back to the `Workers & Pages` overview page at `https://dash.cloudflare.com/[YOUR\_ACCOUNT\_ID]/workers-and-pages`. Click `Create application` > `Pages` > `Connect to Git`
+4. Select the GitLab tab at the top, click the '+ Add account' button, click the GitLab account you want to add, and then click `Authorize` on the modal titled "Authorize Cloudflare Pages to use your account? ".
+5. You should be redirected to the create project page with your GitHub account or organization in the account list.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
3. Go back to the Cloudflare dashboard > **Workers & Pages**.
4. Select **Create application** > **Pages** > **Connect to Git**.
5. Select the GitLab > **+ Add account** > select the GitLab account you want to add > select **Authorize**.
6. You should be redirected to the create project page with your GitHub account or organization in the account list.
```

File: content/pages/platform/git-integration.md [hunk 23]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+6. Attempt to make a new deployment with your project which was previously broken.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md:71]:
```suggestion
7. Attempt to make a new deployment with your project which was previously broken.
```

File: content/pages/platform/git-integration.md [hunk 24]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+## Troubleshooting
+
+If you run into any issues related to deployments failing, check your project dashboard to see if there are any SCM installation warnings listed as shown in the screenshot below.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md:75]:
```suggestion
If you run into any issues related to deployments failing, check your project dashboard to see if there are any SCM installation warnings listed as shown in the following screenshot.
```

File: content/pages/platform/git-integration.md [hunk 25]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+![Pausing a deployment in the Settings of your Pages project](/images/pages/platform/git.dashboard-error.png)
+
+To resolve any errors displayed in the Cloudflare Pages dashboard, follow the steps listed below.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md:79]:
```suggestion
To resolve any errors displayed in the Cloudflare Pages dashboard, refer to the following instructions.
```

File: content/pages/platform/git-integration.md [hunk 26]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+### `This project is disconnected from your Git account, this may cause deployments to fail.`
+
+To resolve this issue, please follow the steps provided above in the [Reinstalling a Git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, please contact support.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
To resolve this issue, follow the steps provided above in the [Reinstalling a Git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, contact support.
```

File: content/pages/platform/git-integration.md [hunk 27]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+### `Cloudflare Pages is not properly installed on your Git account, this may cause deployments to fail.`
+
+To resolve this issue, please follow the steps provided above in the [Reinstalling a Git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, please contact support.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
To resolve this issue, follow the steps provided above in the [Reinstalling a Git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, contact support.
```

File: content/pages/platform/git-integration.md [hunk 28]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+### `The Cloudflare Pages installation has been suspended, this may cause deployments to fail.`
+
+Go to your GitHub installation settings located at:
+
+* `https://github.com/settings/installations` for individual accounts
+* `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts
+
+Click "Configure" on the Cloudflare Pages application. Scroll down to the bottom of the page and click "Unsuspend" to allow Cloudflare Pages to make future deployments.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
* `https://github.com/settings/installations` for individual accounts.
* `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts.

Select **Configure** on the Cloudflare Pages application. Scroll down to the bottom of the page and select **Unsuspend** to allow Cloudflare Pages to make future deployments.
```

File: content/pages/platform/git-integration.md [hunk 29]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+### `The project is linked to a repository that no longer exists, this may cause deployments to fail.`
+
+You may have deleted or transferred the repository associated with this Cloudflare Pages project. For a deleted repository, you will need to create a new Cloudflare Pages project with a repository that has not been deleted. For a transferred repository, you can either transfer the repository back to the original Git account or you will need to create a new Cloudflare Pages project with the transferred repository.
+
+### `The repository is unable to be accessed, this may cause deployments to fail.`
+
+You may have excluded this repository from your installation's repository access settings.  Go to your GitHub installation settings located at:
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
You may have excluded this repository from your installation's repository access settings.  Go to your GitHub installation settings:
```

File: content/pages/platform/git-integration.md [hunk 30]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+* `https://github.com/settings/installations` for individual accounts
+* `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
* `https://github.com/settings/installations` for individual accounts.
* `https://github.com/organizations/<YOUR_ORGANIZATION_NAME>/settings/installations` for organizational accounts.
```

File: content/pages/platform/git-integration.md [hunk 31]
```diff
@@ -45,3 +45,69 @@ To pause automatic deployments, go to the Pages project's **Settings** > **Build
...
+
+Click "Configure" on the Cloudflare Pages application. Under "Repository access," ensure that the repository associated with your Cloudflare Pages project is included in the list.
```
Comment (deadlypants1973) [inline content/pages/platform/git-integration.md]:
```suggestion
Select **Configure** on the Cloudflare Pages application. Under **Repository access**, ensure that the repository associated with your Cloudflare Pages project is included in the list.
```

**Review by deadlypants1973 — APPROVED** [review]
Approved with suggestions

**Review by CarmenPopoviciu — APPROVED** [review]

**Comment by CarmenPopoviciu** [comment]
much awesomeness! thank you for addressing everything!

**Pull request merged**